spec/system/page_objects/components/interface_color_selector.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module PageObjects
+  module Components
+    class InterfaceColorSelector < PageObjects::Components::Base
+      attr_reader :container
+
+      SELECTOR = ".interface-color-selector"
+
+      def initialize(container)
+        @container = container
+      end
+
+      def available?
+        within(container) { has_css?(SELECTOR) }
```

Comment:
Once `container` is a CSS class, then you can replace all these `within` calls with `find(container).has_css?(SELECTOR)` which will be more resiliant to flakes